### PR TITLE
opt: optimize the judgement of supportsHydration

### PR DESCRIPTION
**Pull request by zh-lx**
## Summary

The variable `supportsHydration` has been judged on [line 1354](https://github.com/zh-lx/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L1354), when the code executes to [line 1402](https://github.com/zh-lx/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L1402), the value of `supportsHydration` must be `true`. So it doesn't need to be judged again.

## How did you test this change?

It does not need to add additional tests.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberBeginWork.new.js [hunk 1]
```diff
@@ -1399,7 +1399,7 @@ function updateHostRoot(current, workInProgress, renderLanes) {
     } else {
       // The outermost shell has not hydrated yet. Start hydrating.
       enterHydrationState(workInProgress);
-      if (enableUseMutableSource && supportsHydration) {
```
Comment (zh-lx) [inline packages/react-reconciler/src/ReactFiberBeginWork.new.js:1402]: The variable `supportsHydration` has been judged on [line 1354](https://github.com/zh-lx/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L1354), when the code executes to [line 1402](https://github.com/zh-lx/react/blob/main/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L1402), the value of `supportsHydration` must be `true`. So it doesn't need to be judged again.

**Comment by gaearon** [comment]
This is not right. You're in the `else` branch, so it could be `false` at this point.

**Comment by gaearon** [comment]
Actually wait, I misread the indentation.

**Comment by gaearon** [comment]
This looks good. This does not optimize anything because the flag is static and gets inlined by the compiler. However, it's nice to remove it to avoid confusion about the outer block of code.

**Pull request merged**